ui/app/AppLayouts/Profile/views/BrowserView.qml

```diff
@@ -140,12 +140,15 @@ SettingsContentBase {
 
             PermissionsListView {
                 id: permissionListView
-                walletStore: root.store.walletStore
                 visible: root.store.walletStore.dappList.count > 0
                 anchors.left: parent.left
                 anchors.right: parent.right
                 anchors.leftMargin: Style.current.padding
                 anchors.rightMargin: Style.current.padding
+
+                dappList: root.store.walletStore.dappList
```

Comment:
I don't understand why not mock the `walletStore` as a whole, and instead add 3 more properties?

Reply:
the `walletStore` isn't really needed, this component is really just taking 1 parameters, the dappList and making this explict makes the whole thing much easier to understand and maintain. all this is moving the logic up, one step at the time, at this stage it's going to this layer and next phase it will go the layer above this